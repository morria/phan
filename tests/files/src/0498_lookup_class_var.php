<?php

namespace X;

class Example {
    const BAR = 2;
}

call_user_func(function() {
    $c = 'X\Example';
    echo strlen(new $c());  // should warn
    echo $c::BAR;
    echo $c::BAD;
});
