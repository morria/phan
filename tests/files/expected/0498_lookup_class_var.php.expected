%s:11 PhanTypeMismatchArgumentInternal Argument 1 (string) is \X\Example but \strlen() takes string
%s:13 PhanUndeclaredConstant Reference to undeclared constant \X\Example::BAD (Did you mean \X\Example::BAR)
